api/database_cluster_restore.go

```diff
@@ -107,6 +138,10 @@ func (e *EverestServer) enforceDBRestoreRBAC(user, namespace, srcBackupName, dbC
 	if err := e.enforce(user, rbac.ResourceDatabaseClusterBackups, rbac.ActionRead, rbac.ObjectName(namespace, srcBackupName)); err != nil {
 		return err
 	}
```

Comment:
I think we also need to update this to use the DB cluster name

Reply:
Because with [EVEREST-1706](https://perconadev.atlassian.net/browse/EVEREST-1706), we will be using DB cluster name for Backups as well, just like we do with Restores in this PR.

So this check will basically never pass, since this permissions will not be associated to a specific backup, but its DB.

Reply:
Right, but EVEREST-1706 is not part of this PR.
I planned to address backups in a follow up PR so it's not mixed together.